File: Ds/308.py
s = input()
t = input()
lastsubstringends = 0
subC = 0
nonC = 0
for i in range(len(s) - len(t) + 1):
    if s[i:i+len(t)] == t:
        if i >= lastsubstringends:
            lastsubstringends = i+len(t)
            nonC += 1
        subC += 1
print(subC)
print(nonC)

File: Ds/308.cpp
#include <iostream>
#include <string>
#define forn(i,n) for(long long i = 0; i < n; i++)
#define fornt(i, n) for(size_t i = 0; i < n; i++)
typedef long long ll;
using namespace std;

string s, t;
size_t laststringends, sc, tc; ll subC, nonC;
int main() {
    cin >> s >> t;
    laststringends = 0;
    sc = s.length();
    tc = t.length();
    subC = 0;
    nonC = 0;
    if(sc >= tc) for (size_t i=0; i <=sc - tc; i++) {
        cerr << "running loop" << endl;
        if (s.substr(i, tc) == t) {
            if (i >= laststringends) {
                laststringends = i + tc;
                nonC++;
            }
            subC++;
        }
    }
    cout << subC << endl;
    cout << nonC << endl;
}